gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/TopologyController.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.gateway.rest;
+
+import io.atomix.utils.net.Address;
+import io.camunda.zeebe.broker.client.api.BrokerClient;
+import io.camunda.zeebe.broker.client.api.BrokerClusterState;
+import io.camunda.zeebe.gateway.protocol.rest.BrokerInfo;
+import io.camunda.zeebe.gateway.protocol.rest.Partition;
+import io.camunda.zeebe.gateway.protocol.rest.Partition.HealthEnum;
+import io.camunda.zeebe.gateway.protocol.rest.Partition.RoleEnum;
+import io.camunda.zeebe.gateway.protocol.rest.TopologyResponse;
+import io.camunda.zeebe.util.VersionUtil;
+import java.util.ArrayList;
+import java.util.Set;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@ZeebeRestController
+public final class TopologyController {
+  private final BrokerClient client;
+
+  @Autowired
+  public TopologyController(final BrokerClient client) {
+    this.client = client;
+  }
+
+  @GetMapping(path = "/topology", produces = "application/json")
+  public TopologyResponse get() {
```

Comment:
:wrench: Generally, if you're just creating types, e.g. `new TopologyResponse`, I would recommend using `var` in front because the type is obvious. It also slightly simplifies refactoring in the future.